.github/workflows/format.yml

```diff
@@ -0,0 +1,29 @@
+name: Format
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+defaults:
+  run:
+    working-directory: javascript
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: Format
+        run: |
+          npm ci
+          npm run prettier
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4.1.2
+        with:
+          commit_message: Apply formatting changes
+          branch: ${{ github.head_ref }}
```

Comment:
If the problem before was that using this reference meant we were looking at master? Doesn't this mean the formatting changes will be landed to master? Can we double check this?

In general, it would probably be better not to try to auto-fix the code here, but have two levels:

a) A commit hook to auto-format code LOCALLY (or a script that can be run to do so)
b) A CHECK in CI that will prevent a PR from landing if not formatted correctly, but won't actually attempt to change code.

Reply:
It was stupid of me to write auto-fix code script in a check. I have added in another script `format` which only checks for formatting error and if there are some issues exits with error code `1`.